manifests/reposync-crd.yaml

```diff
@@ -87,9 +87,11 @@ spec:
                     - none
                     type: string
                   branch:
-                    description: branch is the git branch to checkout. The field is
-                      deprecated. Use `revision` instead. If both `branch` and `revision`
-                      are defined, `revision` takes precedence over `branch`.
+                    description: branch is the git branch to sync from. The field
```

Comment:
Behavior updated. It doesn't default to the git server's default branch, but default to `master` for backward compatibility,

Reply:
Wait, what? It defaults to master and not the server default? 
What if the server default is main? It'll just error?

Reply:
It'll error just like today.
The change is to minimize breaking changes.

Reply:
I would be really surprised if there were anyone out there who has both a `main` and a `master` branch and had `main` set to the server default yet still wanted to use the `master` branch from config sync without specifying it.

But I guess if such a scenario did exist, it would start syncing a different branch after they upgraded, which would be bad.

It sucks that we can't improve the default behavior just to avoid breaking a use case that probably doesn't exist.

Reply:
Can we make a v3 API and fix this yet?

Reply:
`make a v3 API` you mean updating git-sync v3 to not set default to `master` if the branch is not specified?
If so, that is also considered a breaking change to git-sync. Moreover, git-sync v3 is in a maintenance mode, I don't think such behavioral change would be accepted.

Reply:
I think Karl means to make a v3 Config Sync API

Reply:
We don't have a v1 or v2 yet. The current Config Sync API is still v1beta1.

Reply:
My understanding (correct me if I'm wrong) is that he's informally referring to monorepo as v1 and multirepo as v2

Reply:
Right. We can't actually change much if we go from v1beta1 to v1. So a "v3" would be more like a brand new custom resource, like Package. Then we could fix field names and behaviors.